### PR TITLE
Allow to configure merchant id, test url and credentials when running the tests

### PR DESCRIPTION
**Pull request by alenad**

### CODE REVIEW

File: lib/litle/api.rb [hunk 1]
```diff
@@ -4,9 +4,10 @@ class PaymentPlugin < ::Killbill::Plugin::ActiveMerchant::PaymentPlugin
 
       def initialize
         gateway_builder = Proc.new do |config|
-          ::ActiveMerchant::Billing::LitleGateway.new :login => config[:login],
+          ::ActiveMerchant::Billing::LitleGateway.new :login => config[:username],
```
Comment (pierre) [inline lib/litle/api.rb]: Why this change?

File: spec/litle/remote/integration_spec.rb [hunk 2]
```diff
@@ -11,8 +11,26 @@
     ::Killbill::Litle::LitleResponse.delete_all
     ::Killbill::Litle::LitleTransaction.delete_all
 
-    @plugin = build_plugin(::Killbill::Litle::PaymentPlugin, 'litle')
-    @plugin.start_plugin
+    Dir.mktmpdir do |dir|
+      file = File.new(File.join(dir, 'litle.yml'), 'w+')
+      file.write(<<-eos)
+:litle:
```
Comment (pierre) [inline spec/litle/remote/integration_spec.rb]: Could we share the `before(:each)` between suites?

File: lib/litle/api.rb [hunk 3]
```diff
@@ -4,9 +4,10 @@ class PaymentPlugin < ::Killbill::Plugin::ActiveMerchant::PaymentPlugin
...
                                                       :password => config[:password],
-                                                      :merchant_id => config[:merchant_id]
+                                                      :merchant_id => config[:merchant_id],
+                                                      :secure_page_url => config[:secure_page_url]
```
Comment (pierre) [inline lib/litle/api.rb]: Why is this now needed? It should only be needed in `application.rb` for PayPage.

File: lib/litle.rb [hunk 4]
```diff
@@ -27,6 +27,14 @@
 module ActiveMerchant #:nodoc:
   module Billing #:nodoc:
     class LitleGateway < Gateway
+      def initialize(options={})
+        if options.has_key?(:secure_page_url)
+          self.test_url = options[:secure_page_url]
```
Comment (pierre) [inline lib/litle.rb]: `test_url` is different than the `secure_page_url`. The former is the API endpoint, not necessarily the same as the PayPage host.

File: lib/litle/api.rb (hunk 1 above)
Comment (alenad) [inline lib/litle/api.rb]: Because the config file uses that key:
https://github.com/killbill/killbill-litle-plugin/blob/master/litle.yml#L4

File: lib/litle/api.rb (hunk 3 above)
Comment (alenad) [inline lib/litle/api.rb]: Vantiv/Litle have a sandbox and prelive environments, so I needed to make the url configurable.

File: spec/litle/remote/integration_spec.rb (hunk 2 above)
Comment (alenad) [inline spec/litle/remote/integration_spec.rb]: Sure

File: lib/litle.rb (hunk 4 above)
Comment (alenad) [inline lib/litle.rb]: I can rename the key to :test_url

File: lib/litle/api.rb (hunk 1 above)
Comment (pierre) [inline lib/litle/api.rb]: Looks like a typo in the file itself, see https://github.com/killbill/killbill-litle-plugin/commit/f5bbfd52b8d657a72267f2d782df37b750649b1a#diff-2fe9adc42378a48d1656caeb98453f11 (seems better to keep `login`).

Comment (alenad) [inline lib/litle/api.rb]: Ok I'll fix the config instead.

File: litle.yml [hunk 5]
```diff
@@ -1,15 +1,15 @@
 :litle:
   - :account_id: "USD"
     :merchant_id: "your-merchant-id-USD"
-    :username: "your-username"
+    :login: "your-username"
     :password: "your-password"
-    :secure_page_url: "https://request-prelive.np-securepaypage-litle.com"
```
Comment (pierre) [inline litle.yml]: You need to keep this so we can still test PayPage locally (`rackup -Ilib -E test`)

File: litle.yml [hunk 6]
```diff
@@ -1,15 +1,15 @@
 :litle:
   - :account_id: "USD"
     :merchant_id: "your-merchant-id-USD"
-    :username: "your-username"
+    :login: "your-username"
```
Comment (pierre) [inline litle.yml]: You could switch to `ENV` variables here, similar to what is done in [CyberSource](https://github.com/killbill/killbill-cybersource-plugin/blob/master/cybersource.yml#L4) for instance.

That way, no need for any changes in tests?

**Comment by alenad** [comment]
@pierre updated the pr

File: litle.yml [hunk 7]
```diff
@@ -1,16 +1,20 @@
...
-    :merchant_id: "your-merchant-id-USD"
-    :username: "your-username"
-    :password: "your-password"
+    :merchant_id: "<%= ENV['LITLE_MERCHANT_ID'] %>"
+    :login: "<%= ENV['LITLE_LOGIN'] %>"
+    :password: "<%= ENV['LITLE_PASSWORD'] %>"
+    :test_url: "<%= ENV['LITLE_TEST_URL'] %>"
     :secure_page_url: "https://request-prelive.np-securepaypage-litle.com"
     :paypage_id: "litle-paypage-id-USD"
+    :test: "<%= ENV['LITLE_TEST'] || false %>"
```
Comment (pierre) [inline litle.yml]: ⚠️ Default should be `true` (this `litle.yml` is only used in tests anyways)!

File: litle.yml [hunk 8]
```diff
@@ -1,16 +1,20 @@
...
-    :merchant_id: "your-merchant-id-EUR"
-    :username: "your-username"
-    :password: "your-password"
+    :merchant_id: "<%= ENV['LITLE_MERCHANT_ID'] %>"
+    :login: "<%= ENV['LITLE_LOGIN'] %>"
+    :password: "<%= ENV['LITLE_PASSWORD'] %>"
+    :test_url: "<%= ENV['LITLE_TEST_URL'] %>"
     :secure_page_url: "https://request-prelive.np-securepaypage-litle.com"
     :paypage_id: "litle-paypage-id-EUR"
+    :test: "<%= ENV['LITLE_TEST'] || false %>"
```
Comment (pierre) [inline litle.yml]: Same here.

**Comment by alenad** [comment]
@pierre updated the pr

**Comment by pierre** [comment]
Thanks!

**Comment by pierre** [comment]
@alenad [Travis](https://travis-ci.org/killbill/killbill-litle-plugin/builds) started failing after the merge. Could you check if it's related to your changes? Do tests pass if `test_url` isn't set?

**Comment by alenad** [comment]
@pierre I just added defaults in config and submitted #6

File: lib/litle.rb [hunk 9]
```diff
@@ -27,6 +27,14 @@
 module ActiveMerchant #:nodoc:
   module Billing #:nodoc:
     class LitleGateway < Gateway
+      def initialize(options={})
+        if options.has_key?(:test_url)
```
Comment (pierre) [inline lib/litle.rb]: Instead of forcing some defaults, can't we simply change this to `unless options[:test_url].blank?`?

Comment (alenad) [inline lib/litle.rb]: If I remove the defaults from `litle.yml` and make this change, then I get the same error as Travis. The reason is that the `commit` method in `litle.rb` accesses the `test_url` and passes it to `ssl_post`, but the url will be nil:
```
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/activemerchant-1.55.0/lib/active_merchant/connection.rb:67:in `request'
/Users/adudzinskaya/.rvm/rubies/jruby-1.7.22/lib/ruby/1.9/benchmark.rb:295:in `realtime'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/activemerchant-1.55.0/lib/active_merchant/connection.rb:60:in `request'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/activemerchant-1.55.0/lib/active_merchant/network_connection_retries.rb:23:in `retry_exceptions'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/activemerchant-1.55.0/lib/active_merchant/network_connection_retries.rb:46:in `retry_network_exceptions'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/activemerchant-1.55.0/lib/active_merchant/network_connection_retries.rb:21:in `retry_exceptions'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/activemerchant-1.55.0/lib/active_merchant/connection.rb:54:in `request'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/activemerchant-1.55.0/lib/active_merchant/posts_data.rb:65:in `raw_ssl_request'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/activemerchant-1.55.0/lib/active_merchant/posts_data.rb:39:in `ssl_request'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/activemerchant-1.55.0/lib/active_merchant/posts_data.rb:35:in `ssl_post'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/activemerchant-1.55.0/lib/active_merchant/billing/gateways/litle.rb:294:in `commit'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/activemerchant-1.55.0/lib/active_merchant/billing/gateways/litle.rb:120:in `store'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/killbill-8.3.0/lib/killbill/helpers/active_merchant/gateway.rb:71:in `method_missing'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/killbill-8.3.0/lib/killbill/helpers/active_merchant/gateway.rb:28:in `store'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/killbill-8.3.0/lib/killbill/helpers/active_merchant/payment_plugin.rb:160:in `add_payment_method'
/Users/adudzinskaya/projects/killbill-litle-plugin/lib/litle/api.rb:119:in `add_payment_method'
/Users/adudzinskaya/.rvm/gems/jruby-1.7.22/gems/killbill-8.3.0/lib/killbill/helpers/active_merchant/killbill_spec_helper.rb:28:in `create_payment_method'
``

Comment (alenad) [inline lib/litle.rb]: I still want to add the `unless options[:test_url].blank?` change since it explicitly deals with empty strings.

Comment (pierre) [inline lib/litle.rb]: > If I remove the defaults from `litle.yml` and make this change, then I get the same error as Travis. The reason is that the `commit` method in `litle.rb` accesses the `test_url` and passes it to `ssl_post`, but the url will be nil

I don't understand -- shouldn't it be set by [ActiveMerchant](https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/litle.rb#L8)?

How did it work before?

Comment (alenad) [inline lib/litle.rb]: I made the change to test with prelive environment. I think setting `test_url` in the yml was ignored until then. It's a [class attribute ](https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateway.rb#L134 )in activemerchant not an option
